Add pt-PT language

Leinad4Mind opened the pull request.

Leinad4Mind committed:
Add pt-PT translation

Copy locale/pt_BR.ts to locale/pt_PT.ts and edit.

vedgy commented:
Please insert the line `    locale/pt_PT.ts \` just under the line `    locale/pt_BR.ts \` in *goldendict.pro*.

I just did that and tried to build GoldenDict. The result is an error:

> lrelease error: Parse error at ../locale/pt_PT.ts:278:27: Expected ''' or '"', but got '[a-zA-Z]'.
make: *** [Makefile:1565: locale/pt_PT.qm] Error 1

Can you build GoldenDict locally to catch and fix this and possible other such errors?

Leinad4Mind commented:
Done, added the locale. Should be all good now.

vedgy commented:
The issue on the line 278, column 27 in the file *pt_PT.ts* is not fixed. You'll see this error (and possibly others after fixing it) if you run `lupdate`, and build GoldenDict locally (or run `lrelease` manually).

Leinad4Mind commented:
Sorry. I am not able to run this locally.

vedgy commented:
Do you not have Qt (`lupdate` and `lrelease`) installed?

You can start by fixing the line 278 and the same issue on the line 283. In the file pt_BR.ts the lines are equal to:
```xml
        <translation type="unfinished"></translation>
```

And in pt_PT.ts:
```xml
        <translation type=Frase não encontrada></translation>
        <translation type=%1 de %2 encontrados></translation>
```

Since the translation is now finished, the `type` attribute should be removed entirely and the translations always go between `>` and `<` (in the `translation` tag). Did you edit the .ts file by hand? I just searched for the `type=[^"]` regex in the file pt_PT.ts and found nothing else like that. So there is a good chance fixing these two lines would fix the build.

vedgy committed:
Fix Forvo language code examples in Portuguese translations

The Forvo language list does not contain '_':
https://forvo.com/languages-codes/

The Forvo API allows to specify the country separately, but GoldenDict
does not support that:
https://api.forvo.com/documentation/word-pronunciations/

vedgy commented on locale/pt_PT.ts:
```diff
@@ -0,0 +1,5213 @@
+    <message>
+        <location filename="../language.cc" line="186"/>
+        <source>Portuguese</source>
+        <translation>Português brasileiro</translation>
```
This translation is wrong both here and in the source file *locale/pt_BR.ts*. Here how it looks in the GoldenDict UI:
![portuguese-wrong-lang-translation](https://github.com/user-attachments/assets/ba0af088-0148-4abf-8dda-8402c0827e08)

The flag is enough to disambiguate the language variants. And that's how it works for other languages. Would the single word `Português` be correct for both Portuguese language variants?

Commodore64user replied:
`Português` and `Português brasileiro` should suffice. another option could be `Português PT` and `Português BR`

vedgy replied:
The problem with `Português brasileiro` is that each language name in the combobox is translated to the currently selected language. So the two rows cannot be different as long as the language is the same. And there is no separate Brazilian language in *language.cc*:
```cpp
  addEntry( "pt", "Portuguese", QCoreApplication::translate( "Language", "Portuguese" ) );
```
Country variants are disambiguated via the flag icons.

So I was asking whether `Português` would be correct as the translation of the word "Portuguese" to both language variants.

vedgy replied:
Done.

vedgy reviewed (changes requested):
Fixed the `type="unfinished"` errors myself, as well as wrong replacements in the `<source>` tags (translators must not modify the contents of these tags), and force-pushed. Also fixed a Forvo translation bug in both Portuguese translations in a separate commit.

I think this PR can be merged once we fix the remaining wrong translation of the word "Portuguese".

vedgy committed:
Fix translations of "Portuguese" to Portuguese

The translation of "Portuguese" to Brazilian Portuguese has been
"Português brasileiro" for years. This was fine until a recent commit
added a European Portuguese translation and copied this particular
translation verbatim. So now there are two consecutive items with the
text "Português brasileiro" in the "Interface language" combobox in
GoldenDict Preferences if one of the Portuguese variants is the current
interface language.

Each language name in the "Interface language" combobox is translated to
the same current interface language. Therefore, the texts of combobox
items that represent different variants of the same language cannot be
different. So change both translations to "Português" and let the
national flags disambiguate the language variants as is the case for
other languages.

vedgy reviewed (approved):
Looks good to me now. @Abs62, OK if I merge this?

Abs62 commented:
@vedgy, OK.
